### DISCUSSION TITLE
can't generate ATP in tutorial

### DISCUSSION THREAD

**Discussion by bilalmoiz93** [SysBioChalmers/Human-GEM#687] [Q&A]
Hello,

I am trying to work through the tutorial. I can solve for biomass with no problem. However, when trying to solve for ATP I return null flux and a message saying the problem is infeasible.  Any suggestions?

>> load('Human-GEM.mat')
>> sol = solveLP(ihuman)

sol = 

  struct with fields:

         x: [13085×1 double]
         f: -138.4666
      stat: 1
       msg: 'Optimal solution found'
    sPrice: [8499×1 double]
     rCost: [13085×1 double]

>> ihuman = setParam(ihuman, 'obj', 'MAR03964', 1);
>> ihuman = setExchangeBounds(ihuman, 'glucose', -1);
>> sol = solveLP(ihuman)

sol = 

  struct with fields:

       x: []
       f: []
    stat: -1
     msg: 'The problem is infeasible'

>> ihuman = setExchangeBounds(ihuman, {'glucose', 'O2'}, [-1, -1000]);
>> sol = solveLP(ihuman)

sol = 

  struct with fields:

       x: []
       f: []
    stat: -1
     msg: 'The problem is infeasible'

**Comment by mihai-sysbio** [comment]
Hi @bilalmoiz93! We hope you found the guide easy to follow. If you have any feedback please do send it this way.

Just to confirm, which solver are you using, and which version of Human-GEM?

**Reply by bilalmoiz93** [reply]
Hello.

Yes! It's pretty straightforward.

I am using the latest version I believe. I only downloaded it a few hours ago and I think it is version 1.16.

For the solver, I have Gurobi installed but when loading RAVEN, it says that Gurobi has failed to load. However, I can run Gurobi with COBRA and COBRApy.

**Reply by haowang-bioinfo** [reply]
@bilalmoiz93 welcome to use Human-GEM

attempted to reproduce the error by using the code you provided, and everything is fine in my side, i.e. ATP can be generated with Human-GEM v1.16.0 and Gurobi v9.5

as @mihai-sysbio suggested, please update your solver, e.g. Gurobi, and see what happens

**Reply by bilalmoiz93** [reply]
Hello

Thank you that worked! Unfortunately now I am running into a seperate problem for the ftInit model.  First, it is taking well over 60 seconds to run. When running with the gtex_data I get the following error

I am getting the following error:
```

Optimal solution found (tolerance 1.00e-12)
Best objective 2.000000000000e+00, best bound 2.000000000000e+00, gap 0.0000%
task: 3
MAR04028
[ER] Aerobic rephosphorylation of CTP: Added 1 reaction(s), 10 reactions added in total
Error using dispEM
One or more equations contain metabolites that are not in model.metNames. Set allowNewMets to true to allow this function to add
metabolites or use addMets to add them before calling this function


Error in addRxns (line 637)
            dispEM(EM);

Error in ftINITFillGapsForAllTasks (line 293)
            tModel=addRxns(tModel,rxn,3);

Error in ftINIT (line 364)
        [outModel,addedRxnMat] = ftINITFillGapsForAllTasks(initModelNoExc,refModelNoExc,[],true,min(rxnScores2nd,-0.1),prepData.taskStruct,paramsFT,verbose);
```

when running with my own custom transcriptomic data, i receive this error:

`
Error using dispEM
Could not find all outputs in "[BS] Protein synthesis from AAs" in either model


Error in ftINITFillGapsForAllTasks (line 190)
                    dispEM(EM);

Error in ftINIT (line 364)
        [outModel,addedRxnMat] =
        ftINITFillGapsForAllTasks(initModelNoExc,refModelNoExc,[],true,min(rxnScores2nd,-0.1),prepData.taskStruct,paramsFT,verbose);

Error in GSE226605_model_construction_7_31_2023 (line 100)
model1 = ftINIT(prepData,data_struct.tissues{1},[],[],data_struct,Listofmets1,getHumanGEMINITSteps('1+0'),false,true)
`

**Reply by mihai-sysbio** [reply]
From what I gather, you're following the `ftINIT` guide by using the files in Zenodo. In the guide, there is a note in ftINIT that regarding `version 1.12.0 of Human-GEM`. Since that release, almost Human-GEM has seen many curations, including the [removal of almost 100 reactions](https://github.com/SysBioChalmers/Human-GEM/compare/v1.12.0...v1.16.0#diff-091a766547b9965ac6c46d2166cc5b2c191018a80b8c3175b64ce16ddf6fae57). Still, I've ran the tutorial with the Zenodo-provided datasets without errors using RAVEN version `2.8.3`:
```
Optimal solution found (tolerance 1.00e-12)
Best objective 4.000000000000e+00, best bound 4.000000000000e+00, gap 0.0000%
task: 57
```

Could you provide a bit more info what versions of the dependencies you are using?
